bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/StyledText.java

```diff
@@ -1593,31 +1593,31 @@ void createCaretBitmaps() {
 		leftCaretBitmap.dispose();
 	}
 	int lineHeight = renderer.getLineHeight();
-	leftCaretBitmap = new Image(display, caretWidth, lineHeight);
-	GC gc = new GC (leftCaretBitmap);
-	gc.setBackground(display.getSystemColor(SWT.COLOR_BLACK));
-	gc.fillRectangle(0, 0, caretWidth, lineHeight);
-	gc.setForeground(display.getSystemColor(SWT.COLOR_WHITE));
-	gc.drawLine(0,0,0,lineHeight);
-	gc.drawLine(0,0,caretWidth-1,0);
-	gc.drawLine(0,1,1,1);
-	gc.dispose();
+	final ImageGcDrawer imageGcDrawer = (gc, width, height) -> {
```

Comment:
I would have inlined the lambda, which also avoids finding a variable name for it.

In hindsight and with this example in mind, I think it would the code would have be a bit nicer if the `ImageGcDrawer` argument would be the last of the new constructor. But unfortunately this ship has sailed 😞